.travis.yml

```diff
@@ -29,6 +30,38 @@ jobs:
+        - secure: sDNv5zI4GfrSMODSdRJ+6NbTh7ppg9eLFGlfTS/hneOBF7Zgt9Yi8P71+OehiAqPOGuXe5L6nP3CqgcCFQzl2oUTgl+aU8ZQFSL4nTX5cstc0T06yqMuyzSr9dGbZFYP3Hb27NpJk7KVe/ddOfTcL/MmEsOJqMuzVvklL7GzuxbjlVm6O1Rhc94lsEWctPaS337yOlrc5CKAxFuTl3ohou3Tode9MuZeVd4XKcxrhfZl/8BAYtfV8wim8hwYDXKTBjcOfB7qd8+3Un4tYAXMGjx5wYKOVOoihuim/7sDn0V+YxHKFhXJp8lWa7oITQaGhlMHGNSCNPdQ0nxnntfdKUtD2rxhAjTUXhoaF+bk4mrjRGxgXN80P9rrQjSCN0JyjWx4tnWj2ivjmhGHQZl4swvMp7k9Sikyj0f9KevtsotQBYAQEzTuzPi0kf9uDiZ05/fnttsHpYKw+BpWqbzl5NBQAuJQuoRREwOGOS4CSpybRhS9Bx/i2+lgfT5NXhr76KWD/uN5KurZvuBgbVzdWSXxq4zh6vM2R9iyvgsCN1TzpU/JXfnQDHC8OJMRDqX7pGoYzwEnq6jatRf8ED7yFm86a5hW1u7bWgIgC3TrkCTXcyZvh8lOKl647i9cxCjLfy411PFim8Ic2UY5jMAEMfVwpLV3HcXbZ7ye8HK/bWI=
+        - secure: HGKRJnJ2Y/JURFKWKWNtFpOMk1u99WIest+2SbuDBlDSVEtVWmUjUpAjz+TdaFt1XIaTANiQrHJdmsc4qgQW4QOhqjpOa4pm3CY74OfG7BYsBsaE5Ja8AiCYoiJ57VFejNQQdUMAzV8GWBjpX0CHbSwsG38ObyU6msql+DhLQXtyZZgbVXHq1DwrCAFB3xDazzePzcRuVikIZ70RoM5qa7VLxfWhI1oqHp31il4x3lr+vPYjWDqNulzXXv+8mD9A/GpEtAlDRJ2u2kH/3lSVNkIncUGKbb4Z2OiNUA97V7COLa31lhuDGSTAxVbf9xM9nnEHz/np3z9IHBZMvlCOCIBSPwgClCkUqEp0efKt6OufuAV/TkyFi0Llp4jeaTGeRRfPYF6hAZBbAR+2mOlyom6L6MBw0jrdnJq7euqdS/k/CdxkyMaXGGjqridGRrsRwZ//2ULcXbMgmKehSaqez/1ss4UtnV6LQqqbQQbZQppywz/j9bUoS8XGxaOAfhATRxOLbqQgKYmRD5YnJB+leVwFg3/YpTAiReEMrSwbyEN8OcAenHZtdCg+JwV/ZgobcerBTK0vXcWWlQgttqQEyT41qRU5kvkDfCc1UFDYWkVfaaRs0t+l4W+2YmtzIpDQk081JqFZHz6fXbadAIQu+J8Igap6bp9Y86eMUlLIwc4=
+      install:
+        - curl -sL https://aka.ms/InstallAzureCLIDeb | sudo bash
+      before_script:
+        - az login --service-principal --username "$USER_NAME"  --password "$PASSWORD" --tenant "$TENANT" > /dev/null
+        - ./buildZip.sh
+        - az storage fs create --name "$STORAGE_ACCOUNT_DIR_NAME" --account-name "$STORAGE_ACCOUNT_NAME" --account-key "$STORAGE_ACCOUNT_KEY" --public-access file
+        - az storage fs file upload --file-system "$STORAGE_ACCOUNT_DIR_NAME" --account-name "$STORAGE_ACCOUNT_NAME" --account-key "$STORAGE_ACCOUNT_KEY" --source "publish.zip" --path "dynatrace-azure-log-forwarder.zip"
+        - az storage fs file upload --file-system "$STORAGE_ACCOUNT_DIR_NAME" --account-name "$STORAGE_ACCOUNT_NAME" --account-key "$STORAGE_ACCOUNT_KEY" --source "deployment/dynatrace-azure-forwarder.json" --path "dynatrace-azure-forwarder.json"
+        - az group create --name "$RESOURCE_GROUP_NAME" --location westeurope
+      script:
+        - ./deployment/dynatrace-azure-logs.sh --deployment-name "e2e-${TRAVIS_BUILD_ID}" --resource-group "$RESOURCE_GROUP_NAME" --target-url "$TARGET_URL" --target-api-token "$TARGET_API_TOKEN" --target-paas-token "$TARGET_PAAS_TOKEN" --event-hub-name "logs-forwarding-e2e-eh" --event-hub-connection-string "$EH_CONNECTION_STRING" --repository-release-url "https://${STORAGE_ACCOUNT_NAME}.blob.core.windows.net/${STORAGE_ACCOUNT_DIR_NAME}/"
```

Comment:
If the build artifact was shared between stages, you would not need the additional option to define other package sources.

Reply:
For Azure deployment script is impossible. The Azure CLI command we use to create deployment requires ARM json from some URL. 
I don't want to change the deployment script only for E2E test.